src/support/steps/lcp-beacon-script.ts

```diff
@@ -147,4 +148,78 @@ Then('lcp and atf should be as expected in {string}', async function (this: ICus
     }
 
     expect(truthy, failMsg).toBeTruthy();
-});
\ No newline at end of file
+});
+
+/**
+ *
+ */
+When('I go to {string} in {string}', async function (this: ICustomWorld, page, formFactor: string) {
+    let viewPortWidth: number = 1600,
+        viewPortHeight: number = 700;
+
+    // Set page to be visited in mobile.
+    if ( formFactor === 'mobile' ) {
+        viewPortWidth = 393;
+        viewPortHeight = 830;
+    }
+
+
+    await this.page.waitForLoadState('load', { timeout: 100000 });
+
+    await this.page.setViewportSize({
+        width: viewPortWidth,
+        height: viewPortHeight
+    });
+
+    const tablePrefix: string = await getWPTablePrefix();
+
+    await this.utils.visitPage(page);
+
+    // Wait for 6 seconds before fetching from DB.
+    await this.page.waitForTimeout(6000);
```

Comment:
We can borrow @Miraeld implementation from this [PR](https://github.com/wp-media/wp-rocket-e2e/pull/75)
```suggestion
    // Wait the beacon to add an attribute `beacon-complete` to true before fetching from DB.
        await this.page.waitForFunction(() => {
            const beacon = document.querySelector('[data-name="wpr-lcp-beacon"]');
            return beacon && beacon.getAttribute('beacon-completed') === 'true';
        });
```